src/decorators/transport/TransportDecoratorExtension.ts

```diff
@@ -31,6 +31,11 @@ export function TransportDecorated<T extends BaseMessageConstructor>(Base: T) {
       // transport is thread but threadId is either missing or doesn't match. Return false
       return false
     }
+
+    public hasAnyReturnRoute() {
+      const returnRoute = this.transport?.returnRoute
```

Comment:
Not really a fan of this method in addition to the already exiting method. But I see the limitation of the first method

Reply:
Hmm, I'm not a fan of that either. We should perhaps refactor both to better communicate what's going on there 🤔  I made a mistake when using the other method inside `MessageSender` as you found.